indexer/sql/src/scalars.rs

```diff
@@ -0,0 +1,52 @@
+use {
+    async_graphql::{Scalar, ScalarType, Value},
+    grug_types::Timestamp,
+    sqlx::types::chrono::{DateTime as ChronoDateTime, NaiveDateTime},
+};
+
+/// A custom DateTime scalar that ensures ISO8601 compliance with timezone information
+#[derive(Clone, Debug)]
+pub struct Iso8601DateTime(pub NaiveDateTime);
+
+#[Scalar(name = "DateTime")]
+impl ScalarType for Iso8601DateTime {
+    fn parse(value: Value) -> async_graphql::InputValueResult<Self> {
+        match value {
+            Value::String(s) => {
+                // Try parsing as RFC3339 first (with timezone)
+                if let Ok(dt) = ChronoDateTime::parse_from_rfc3339(&s) {
+                    Ok(Self(dt.naive_utc()))
+                } else if let Ok(dt) = NaiveDateTime::parse_from_str(&s, "%Y-%m-%dT%H:%M:%S") {
+                    // Parse basic ISO format without timezone, assume UTC
+                    Ok(Self(dt))
+                } else if let Ok(dt) = NaiveDateTime::parse_from_str(&s, "%Y-%m-%dT%H:%M:%S%.f") {
+                    // Parse ISO format with microseconds but no timezone, assume UTC
+                    Ok(Self(dt))
+                } else {
+                    Err(async_graphql::InputValueError::custom(
+                        "Invalid datetime format",
+                    ))
+                }
+            },
+            _ => Err(async_graphql::InputValueError::custom("Expected string")),
+        }
```

Comment:
@penso This logic is duplicate as in `serde_iso8601.rs`. I'm going to reuse the function to avoid the duplication. Also I'm going to use the early return pattern to make the code look nicer.